### PR TITLE
44 add admin outlet checkauth routes

### LINKED ISSUE #44: add admin [outlet, checkAuth, routes]

### PR DESCRIPTION
**Pull request by Nour-Agha99**

### CODE REVIEW

**Review by mobatra — APPROVED** [review]

File: src/components/LoginForm/FormAdmin.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  } = useForm<FormData>({ resolver: zodResolver(loginSchema), mode: 'onBlur' });
+
+  const { isLoading, mutate, error } = useMutation((data: FormData) => authRoutes.login(data), {
+    onSuccess: (data) => {
+      localStorage.setItem('user', JSON.stringify(data.data.data.user));
+      if (data.data.data.user.isAdmin) {
+        navigate('/admin');
+      } else {
+        toast.error('This page just for Admin');
+        navigate('/admin');
```
Comment (karamzomlot) [inline src/components/LoginForm/FormAdmin.tsx:43]: I think here you shouldn't navigate to `/admin` just show the alert

File: src/components/LoginForm/FormUser.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({ resolver: zodResolver(loginSchema), mode: 'onBlur' });
+  const { isLoading, mutate, error } = useMutation((data: FormData) => authRoutes.login(data), {
+    onSuccess: (data) => {
+      localStorage.setItem('user', JSON.stringify(data.data.data.user));
+
+      // eslint-disable-next-line no-unused-expressions
+      data.data.data.user ? navigate('/academy') : '';
```
Comment (karamzomlot) [inline src/components/LoginForm/FormUser.tsx:40]: What if the admin trying to log in from this page?

e.g.:
```ts
if (data.data.data.user.isAdmin) {
  navigate('/admin');
} else {
  toast.error('This page just for Admin');
}
```

**Review by karamzomlot — CHANGES_REQUESTED** [review]

File: src/components/LoginForm/FormAdmin.tsx (hunk 1 above)
Comment (Nour-Agha99) [reply src/components/LoginForm/FormAdmin.tsx:43]: coz iam handleing delete token in the provider checkAuthAdmin

File: src/components/LoginForm/FormUser.tsx (hunk 2 above)
Comment (Nour-Agha99) [reply src/components/LoginForm/FormUser.tsx:40]: but we are dealing, the admin can not use this page ( Academy/login ) to signIn!!

File: src/components/LoginForm/FormAdmin.tsx (hunk 1 above)
Comment (karamzomlot) [reply src/components/LoginForm/FormAdmin.tsx:43]: Aha I got it, Okay
Good Job man🔥

File: src/components/LoginForm/FormUser.tsx (hunk 2 above)
Comment (karamzomlot) [reply src/components/LoginForm/FormUser.tsx:40]: I'm sorry, I think I copied and pasted the wrong code.
What I want to tell you is to handle the error if the admin is trying to log in from the user login page

```ts
if (data.data.data.user.isAdmin) {
  toast.error('Try to login from admin page please ');
} else {
  //Implement your login for the user
}
```

Comment (Nour-Agha99) [reply src/components/LoginForm/FormUser.tsx:40]: done

**Review by KhaledAref13 — APPROVED** [review]

**Review by karamzomlot — APPROVED** [review]

**Pull request merged**